### PR TITLE
Vuefy Job information

### PR DESCRIPTION
**Pull request by OlegZharkov**
Rework another part of `show_params.mako`, this time we move Job Information to Vue.
We're almost there. After this PR, only job dependencies left and we finally can remove that .mako file
this PR depends on https://github.com/galaxyproject/galaxy/pull/10327
Exaple:

image

### CODE REVIEW

**Comment by mvdbeek** [comment]
Can you pull stdout and stderr into the top-level view if they are present ? The extra link is kind-of silly.

**Comment by OlegZharkov** [comment]
> Can you pull stdout and stderr into the top-level view if they are present ? The extra link is kind-of silly.

Sorry, didn't really get it. What did you mean by top level view?

 You didn't like the method or am i missing something? The method is created because of getGalaxyRoot()

File: lib/galaxy/webapps/galaxy/api/security.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import logging
+
+from galaxy.web import (
+    expose_api,
+    require_admin
+)
+from galaxy.webapps.base.controller import (
+    BaseAPIController,
+    UsesVisualizationMixin
+)
```
Comment (dannon) [inline lib/galaxy/webapps/galaxy/api/security.py]:
```suggestion
from galaxy.webapps.base.controller import BaseAPIController
```

File: lib/galaxy/webapps/galaxy/api/security.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+log = logging.getLogger(__name__)
+
+
+class SecurityController(BaseAPIController, UsesVisualizationMixin):
```
Comment (dannon) [inline lib/galaxy/webapps/galaxy/api/security.py]:
```suggestion
class SecurityController(BaseAPIController):
```

Comment (dannon) [reply lib/galaxy/webapps/galaxy/api/security.py]: Don't need that mixin, right?

File: lib/galaxy/webapps/galaxy/api/security.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+from galaxy.webapps.base.controller import (
+    BaseAPIController
+)
+
+log = logging.getLogger(__name__)
+
+
+class SecurityController(BaseAPIController):
+
+    @require_admin
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/security.py]: There is a encode_id in the config controller. Can you put this on that controller instead - and add a matching integration test in test_configuration_decode.py (maybe renaming that file)?

Comment (OlegZharkov) [reply lib/galaxy/webapps/galaxy/api/security.py]: > There is a encode_id in the config controller. Can you put this on that controller instead - and add a matching integration test in test_configuration_decode.py (maybe renaming that file)?

do you mean to move it under `galaxy/lib/galaxy/security/` directory?

Comment (jmchilton) [reply lib/galaxy/webapps/galaxy/api/security.py]: https://github.com/galaxyproject/galaxy/blob/dev/test/integration/test_configuration_decode.py
https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/api/configuration.py#L100

Comment (OlegZharkov) [reply lib/galaxy/webapps/galaxy/api/security.py]: > https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/api/configuration.py#L100

thanks, wasn't aware about this method. This helps a lot

**Comment by OlegZharkov** [comment]
I guess, I will rebase this, to have Selenium testing in GHA

**Comment by mvdbeek** [comment]
Damn, looks like the bibtex test failed. I'll look into that.

File: lib/galaxy/managers/jobs.py [hunk 4]
```diff
@@ -662,3 +662,24 @@ def inputs_recursive(input_params, param_values, depth=1, upgrade_messages=None)
 
     parameters = inputs_recursive(tool.inputs, params_objects, depth=1, upgrade_messages=upgrade_messages)
     return {"parameters": parameters, "has_parameter_errors": has_parameter_errors}
+
+
+def summarize_job_information(trans, job):
+    """Produce a dict-ified version of job information
```
Comment (mvdbeek) [inline lib/galaxy/managers/jobs.py]: Can you use the existing endpoint in api/jobs/job_id and do a separate call to decode ids for admins?

Comment (OlegZharkov) [reply lib/galaxy/managers/jobs.py]: @mvdbeek 
So, I tried and I am missing `copied_from_job_id` and `toold_id`

Comment (mvdbeek) [reply lib/galaxy/managers/jobs.py]: I think it's fine to add them to the full_details view

**Comment by OlegZharkov** [comment]
JobStore could be reused for future work in converting `show_params.mako`

File: lib/galaxy/managers/jobs.py (hunk 4 above)
Comment (OlegZharkov) [reply lib/galaxy/managers/jobs.py]: @mvdbeek 
thanks a lot for helping me offline

**Comment by OlegZharkov** [comment]
Can someone with access remove `WIP` labels here please? So bot can assign 21.01 milestone

**Comment by jmchilton** [comment]
Awesome stuff - thanks so much!

**Comment by OlegZharkov** [comment]
> Awesome stuff - thanks so much!

thanks @jmchilton!

**Pull request merged**